lib/src/modules/task.cc

```diff
@@ -0,0 +1,32 @@
+#include <pxp-agent/modules/task.hpp>
+
+#define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pxp_agent.modules.task"
+#include <leatherman/logging/logging.hpp>
+
+namespace PXPAgent {
+namespace Modules {
+
+namespace fs = boost::filesystem;
+
+// TODO: construct and pass metadata to a new ExternalModule constructor, rather than calling `task metadata`
+// TODO: how do we get the path to task.exe if it's co-located with pxp-agent, rather than in modules dir?
```

Comment:
It might be easier to put it in the modules dir anyway, and instantiate it after other external modules. It'd get skipped if it doesn't respond to metadata. Finding the binary install location for pxp-agent is kind of a pain.